app.js

```diff
@@ -0,0 +1,17 @@
+var express = require('express');
+var app = express();
+var mongoose = require('mongoose');
+
+var bodyParser = require('body-parser');
+var Truck = require('./routes/truckRoutes') (Truck);
```

Comment:
When you require your truckRoutes module in, assign it to a truckRouter variable instead:

`var truckRouter = require('./routes/truckRoutes');`

You don't want to pass in Truck to a function invocation here, because your truckRoutes isn't returning a function that can be called, it's returning an object.